aiida/orm/nodes/node.py

```diff
@@ -885,18 +892,22 @@ def get_incoming(self, node_class=None, link_type=(), link_label_filter=None):
             link type, if None then returns all inputs of all link types.
         :param link_label_filter: filters the incoming nodes by its link label.
             Here wildcards (% and _) can be passed in link label filter as we are using "like" in QB.
+        :param only_uuid: Use the uuid of the nodes instead of the `Node` objects
         """
         if not isinstance(link_type, tuple):
             link_type = (link_type,)
 
         if self.is_stored:
-            link_triples = self.get_stored_link_triples(node_class, link_type, link_label_filter, 'incoming')
+            link_triples = self.get_stored_link_triples(
+                node_class, link_type, link_label_filter, 'incoming', only_uuid=only_uuid)
         else:
             link_triples = []
 
         # Get all cached link triples
         for link_triple in self._incoming_cache:
 
+            if only_uuid:
```

Comment:
Don't quite understand. Has this been resolved?

Reply:
Yes it has been resolved. My original comment is wrong, because in the case of `only_uuid` the link triples from the stored nodes will already only have the uuid, so the conversion of the cached triples needs to go before the check. Which is what you already had.